### ISSUE TITLE
Exceptions with nil for message cause error in v6.1.1

### ISSUE THREAD

**Issue by tylerjj** [airbrake/airbrake-ruby#697]
* Airbrake version: `v6.1.1`
* Ruby version: `2.6.10`
* Framework name & version: `Rails v6.0.5`

### Airbrake config
```ruby
Airbrake.configure do |c|
  c.project_id = *******
  c.project_key = ******
  c.root_directory = Rails.root
  c.logger = Rails.logger
  c.environment = airbrake_environment
  c.ignore_environments = %w[test development]
  c.blocklist_keys = Rails.application.config.filter_parameters
  c.performance_stats = false
end
```
### Description
With the changes made in `v6.1.1`, `airbrake-ruby` is now assuming the `message` attribute of an `exception` is not `nil` when  `notify_airbrake(exception)` is called. 

In our case, we have an `error` coming from the `Stripe` gem where `error.message` is `nil`.
When we call `notify_airbrake(error)`, the following is triggered by the `airbrake-ruby` gem:

#### Airbrake Error
```
NoMethodError: undefined method `split' for nil:NilClass
```
#### Error Source - line 28 https://github.com/airbrake/airbrake-ruby/blob/v6.1.1/lib/airbrake-ruby/nested_exception.rb#L28
```ruby
            def as_json
              unwind_exceptions.map do |exception|
                {   type: exception.class.name,
 28                 message: exception.message.split(RUBY_31_ERROR_HIGHLIGHTING_DIVIDER).first,
                    backtrace: Backtrace.parse(exception) }
              end
            end
```

A band-aid fix was to set the `message` variable `error.instance_variable_set(:@message, "no message") if error.message.nil?` before we call `notify_airbrake(error)`. 

By downgrading to `v6.1.0` the issue was resolved.

**Comment by kyrylo** [comment]
Just curious, what exception is that? If I instantiate a new exception, it always has a message: 

```
irb(main):001:0> RuntimeError.new
=> #<RuntimeError: RuntimeError>
irb(main):002:0> RuntimeError.new.message
=> "RuntimeError"
```

**Comment by tylerjj** [comment]
## Some Background on Stripe Errors ##
This is where the Stripe gem defines its custom errors:
https://github.com/stripe/stripe-ruby/blob/v7.1.0/lib/stripe/errors.rb
You can see that they inherit from `StandardError` but the constructor defaults the `message` to `nil`.

## Context of Stripe Error ##

In our tests, we stubbed a response from Stripe that had a certain response code but we didn’t stub out a message.

Although Stripe Production responses might always provide an error message, they have an object that inherits from `StandardError` that doesn’t require a message and in their own automated tests they initialize the object without one, which is somewhat similar to what we were doing.
https://github.com/stripe/stripe-ruby/blob/v7.1.0/test/stripe/stripe_client_test.rb#L351

That line uses `Stripe::StripeError.new(http_status: 409)` which would produce something like this:
```ruby
error = Stripe::StripeError.new(http_status: 409)
error.message
 => nil
 ```

### How To Recreate The `notify_airbrake` Error ###
Creating a `DummyController`:
```ruby
2.6.10 :001 > 
class DummyController < ApplicationController
  class TestError < StandardError
    def message
      nil
    end
  end

  def test_notify_airbrake
    notify_airbrake(TestError.new)
  end

  def test_airbrake_notify
    Airbrake.notify(TestError.new)
  end
end
2.6.10 :016 > dummy_controller = DummyController.new
```
Testing `notify_airbrake`:
```ruby
2.6.10 :017 > dummy_controller.test_notify_airbrake
Traceback (most recent call last):
        2: from (irb):17
        1: from (irb):9:in `test_notify_airbrake'
NoMethodError (undefined method `split' for nil:NilClass)
```
Testing `Airbrake.notify`
```ruby
2.6.10 :018 > test_controller.test_airbrake_notify
 => #<Airbrake::Promise:0x00007fee1e3eada0 @on_resolved=[], @on_rejected=[], @value={"error"=>"current environment 'development' is ignored"}, @mutex=#<Thread::Mutex:0x00007fee1e3ead00
```
`Airbrake.notify` does not generate the same problem/error that this issue is reporting on.

**Comment by kyrylo** [comment]
It's unfortunate that Stripe violates the type expectation but I think there's more code in the wild that does this. Sorbet to the rescue? 😛 I think the message should default to the classname, so that it's consistent with Ruby's default behavior.

Thanks for the explanation, though.

**Comment by kyrylo** [comment]
I just released v6.1.2 with the fix. Make sure to `bundle update`.
